tests/Fixer/Basic/BracesFixerTest.php

```diff
@@ -3245,10 +3222,10 @@ public static function provideFixCommentBeforeBraceCases(): iterable
     // 2.5+ API
     if (isNewApi()) {
         echo "new API";
-    // 2.4- API
+        // 2.4- API
```

Comment:
Initially, `statement_indentation` supported unindenting these comments only for `braces` backward compatibility. I think we should enable the option when used via `braces` to keep the previous behavior.

Reply:
I think this PR should be reverted completely, as 1 fixed/configured indentation is enough for all usecases. We need to allow both.

Reply:
Is it possible to automatically configure the fixer depending on the used ruleset? Do we have such a mechanism or should we hardcode it somehow?

Reply:
This depends per code usage, thus both must be allowed and no configuration is needed.

Reply:
@Wirone I meant to enable the option when initializing `brace`'s internal fixers, like is already done with some of them:

https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/250d9570d8966ab1a24d92371b03e9f15c2b03e4/src/Fixer/Basic/BracesFixer.php#L224

However I actually agree with @mvorisek here. I think it would be more accurate to simply allow both levels of indentation and only fix to the closest one, e.g.:
```diff
 function test() {
     if ($foo) {
         foo();
-// bar
+    // bar
     } elseif ($bar) {
         bar();
-            // baz
+        // baz
     } else {
         baz();
     }
 }
```

I would keep this behind an option though, e.g. `allow_comments_stuck_to_next_continuous_control_statement` (false by default).

Reply:
Ah, right - configure while proxying 🙂. Yeah, that sounds good.

Reply:
> I think this PR should be reverted completely, as 1 fixed/configured indentation is enough for all usecases. We need to allow both.

@mvorisek 

the logic that allow `// foo` to be treated for next `] else {` was introduced originally without option, you complained to revert it. I reverted back to original behaviour yet that new behaviour was needed by Symfony community is still needed, thus added behind configuration flag.

I do understand that whatever I'm bringing for usecases of my ecosystem is not interesting for you, and you are free to not use it, but please avoid statements like "we don't need this" or "this PR should be reverted". Those are really demotivating.

Reply:
@julienfalque actually, option was introduced for Symfony ecosystem and there, the comment is not "let's use the closest", but very explicit where to put it - on level of next `else` statement. If we know the rule, no need to guess the location to "better, but not sure if proper", we can put to proper (but yeah, with explicit config, as not everyone likes it). Maybe the default could be "put to closest", and still for Sf we have config for "put explicitly for next `else` block".

Or overkill and not worth to have that yet another possibility. If someone wants Sf style, they don't want "get to closest". If someone doesn't want Sf style, they want accurate placing as well. Guessing will guess, but not always hit the target. Giving target explicitly is better imho.

```diff
 function test() {
     if ($foo) {
         foo();
-// bar
+    // bar
     } elseif ($bar) {
         bar();
-            // baz
+    // baz (different than in your example)
     } else {
         baz();
     }
 }
```
```

Reply:
`braces` - I really hope nobody uses it anymore. And all built-in rulesets are already migrated from it.
I suggest to not invest in it unless someone explicitly requires that and cannot migrate away from it

Reply:
Keeping the current option and changing its possible values to something like `always|never|loose` seems like a good compromise to me. Indeed, the `loose` case will likely not be 100% right, but it will allow users to adjust to the expected level of indentation if needed. But yeah maybe really edge case and not worth it.